src/software/thunderscope/gl/graphics/gl_label.py

```diff
@@ -0,0 +1,78 @@
+from PyQt6.QtGui import QFont, QColor
+from pyqtgraph.opengl.GLGraphicsItem import GLGraphicsItem
+from pyqtgraph.Qt import QtCore, QtGui
+
+from typing import Optional
+
+from software.thunderscope.gl.graphics.gl_painter import GLPainter
+from software.thunderscope.constants import Colors
+
+
+class GLLabel(GLPainter):
+    """Displays a 2D text label on the viewport"""
+
+    def __init__(
+        self,
+        parent_item: Optional[GLGraphicsItem] = None,
+        font: QFont = QFont("Roboto", 8),
+        text_color: QColor = Colors.PRIMARY_TEXT_COLOR,
+        offset: tuple[int, int] = (0, 0),
+        text: str = "",
+    ) -> None:
+        """Initialize the GLLabel
+
+        :param parent_item: The parent item of the graphic
+        :param font: The font using to render the text
+        :param text_color: The color for rendering the text.
+        :param offset: The offset (x, y) from the viewport left and top edge
+                       to use when positioning the label.
+                       If x is negative then the x offset is |x| pixels from
```

Comment:
nit: is this necessary? feels like its a bit clearer to just require the offsets to be positive numbers always. from the perspective of someone using this method in the future, the sign flipping feels a bit roundabout

Reply:
not a huge deal tho

Reply:
Lol I followed the design of other components on the same level for consistency. But sure, I will clean this up to make it more understandable.

Reply:
After some testing here, I found out that the easier way to place the label on the top right corner is just to keep this wired sign flipping, because it takes some code to fetch the width of the viewport before passing into this function. Instead, we can get viewport by whatever calls this function, so we probably need to use a negative sign to denote starting from the right side.